index.html

```diff
@@ -5,45 +5,364 @@
+{"num":8,"neighborhood":"Downtown","poverty_rate":26,"avg_family_income":142881},
+{"num":35,"neighborhood":"Naylor Gardens","poverty_rate":17,"avg_family_income":72747},
+{"num":6,"neighborhood":"DuPont Circle","poverty_rate":13,"avg_family_income":177893},
+{"num":20,"neighborhood":"Michigan Park","poverty_rate":11,"avg_family_income":87062},
+{"num":21,"neighborhood":"Edgewood","poverty_rate":17,"avg_family_income":91520},
+{"num":2,"neighborhood":"Columbia Heights","poverty_rate":17,"avg_family_income":87204},
+{"num":7,"neighborhood":"Logan Circle","poverty_rate":12,"avg_family_income":123697},
+{"num":3,"neighborhood":"Shaw","poverty_rate":17,"avg_family_income":101583},
+{"num":27,"neighborhood":"Navy Yard","poverty_rate":34,"avg_family_income":77952}];
+
+var MAX_INCOME = 304995;
+var MAX_POVERTY = 45;
+
+var options = {
+  sorted: false,
+  filtered: false,
+  page:0
+};
+
+
+
+
+function getNumberOfPages() {
+  return Math.ceil(dc_info.length/numberPerPage);
+}
+
+function nextPage() {
+  currentPage += 1;
+  updateTable();
+}
+
+//Sort by neighborhood name and poverty rate in descending order
+
+function compareStrings (a,b) {
+  a = a.toLowerCase();
+  b = b.toLowerCase();
+  return (a < b) ? -1 : (a > b) ? 1: 0;
+}
+dc_info.sort(function (a, b) {
+  return compareStrings(a.neighborhood, b.neighborhood);
+});
+
+//Sort by income and poverty rate
+
+
+function updateTable() {
+
+  var data = dc_info.slice()
+
+
+//Filter by Poverty>20%
+  if (options.filtered) {
+     if (options.filtered === 'filter-p') {
+        data = data.filter(function (a) { 
+          return a.poverty_rate > 20; });
+      }
+}
+//Filter by 6 figure income
+  if (options.filtered) {
+    if (options.filtered === 'filter-i'){
+      data = data.filter(function (a) {
+        return a.avg_family_income > 99999; });
+    }
+}
+
+//Sort
+
+if (options.sorted) {
+    if (options.sorted === 'desc') {
+      data.sort(function (a, b) { return b.avg_family_income - a.avg_family_income; });
+    } else if (options.sorted === 'desc-poverty') {
+      data.sort(function (a, b) { return b.poverty_rate - a.poverty_rate; });
+    }
+ }
+
+//My attempt at Pagination for Next button, not working though
+
+
+
+  if (options.page) {
+    if (options.page === 'next-page'){
+      
+      data = data.slice(function (pageno, max_num) {
+        console.log('3');
+        var pageno = 2;
+        var rows = 10;
+        
+        return data.slice((pageno - 1) * rows, pageno * rows); });
+    }
+  } 
+
+  var table = document.getElementById('data-table');
+  var tbody = table.getElementsByTagName('tbody')[0];
+
+  tbody.innerHTML = '';
+
+  data.forEach(function (d) {
+
+    var row = document.createElement('tr');
+
+    var name = document.createElement('td');
+    name.innerHTML = d.neighborhood;
+    row.appendChild(name);
+
+    var income= document.createElement('td');
+    income.innerHTML = '$' + new Intl.NumberFormat().format(d.avg_family_income);
```

Comment:
🎩 for using `Intl.NumberFormat()`. It's still too new to be supported in a lot of browsers, though, but fortunately, we'll soon be able to replace it with [d3.format()](https://github.com/d3/d3/wiki/Formatting).